controllers/storageautoscaler/scraper.go

```diff
@@ -99,24 +109,47 @@ func filterObjectsForScaling(storageAutoScalerList *ocsv1.StorageAutoScalerList,
 	for _, storageAutoScaler := range storageAutoScalerList.Items {
 		deviceClass := storageAutoScaler.Spec.DeviceClass
 		deviceClassThreshold := storageAutoScaler.Spec.StorageScalingThresholdPercent
-		// get the osd usage from the sync map
-		usage, ok := syncMap.Load(deviceClass)
-		if !ok {
-			log.Error(fmt.Errorf("osd usage not found for device class"), "osd usage not found for device class", "device class", deviceClass)
-		} else if usage == nil {
-			err := fmt.Errorf("osd usage is nil for device class, device class: %s", storageAutoScaler.Spec.DeviceClass)
-			log.Error(err, "device class not found in sync map")
-		} else {
-			deviceClassUsage := usage.(float64)
-			if (deviceClassUsage * 100) >= float64(deviceClassThreshold) {
-				objectList = append(objectList, types.NamespacedName{Namespace: storageAutoScaler.Namespace, Name: storageAutoScaler.Name})
-			}
+		if checkIfScalingRequired(syncMap, deviceClassThreshold, log, deviceClass) {
+			objectList = append(objectList, types.NamespacedName{Namespace: storageAutoScaler.Namespace, Name: storageAutoScaler.Name})
 		}
 	}
 
 	return objectList
 }
 
+func checkIfScalingRequired(syncMap *sync.Map, deviceClassThreshold int, log logr.Logger, deviceClass string) bool {
```

Comment:
Just a simpler name suggestion:
```suggestion
func isScalingRequired(syncMap *sync.Map, deviceClassThreshold int, log logr.Logger, deviceClass string) bool {
```

Reply:
Done

Reply:
Tagged!